src/common/playlists/PlayHeaderActions.jsx

```diff
@@ -78,9 +81,17 @@ const PlayHeaderActions = ({ play }) => {
     }
   };
 
+  const handleEditPlay = () => {
+    const generateLink = `/editplay/${play.github}/${play.slug}`;
+    return navigate(generateLink);
+  };
+
   return (
     <>
       <Like onLikeClick={!loading ? onLikeClick : null} likeObj={likeObj} />
+      <button className="action">
+        <AiFillEdit onClick={handleEditPlay} size={24} className='icon' />
```

Comment:
Shouldn't provide this button if the current user is not the owner of the play
If not logged in, then it should get authenticated first. If the user is not the author, then show an error msg

Reply:
didnt understand. if we only show button if the current user is owner of the play or not means we are assuming user is authenticated. if not then we donot show any button. but also we donot have auth thing seperately so only the like button currently which makes user authenticate. i dont think we should depend on like button to be clicked then user will authenticate then edit button will appear or not checking the owernship and then we allow to edit?

thats how i thought to put edit button on each play which will do auth (if not) and check if the user is owner or not then proceed to edit it otherwise show error message.

looking forward for other thoughts.